vscode-dotnet-sdk-extension/package.json

```diff
@@ -0,0 +1,110 @@
+	"displayName": ".NET SDK Install Tool",
+	"description": "Allows acquisition of the .NET SDK.",
+	"icon": "images/dotnetIcon.png",
+	"version": "0.0.1",
+	"publisher": "ms-dotnettools",
+	"engines": {
+		"vscode": "^1.41.0"
+	},
+	"preview": true,
+	"categories": [
+		"Other"
+	],
+	"keywords": [
+		".NET",
+		".NET Core",
+		"dotnet",
+		".NET SDK"
+	],
+	"activationEvents": [
+		"onCommand:dotnet-sdk.acquire",
+		"onCommand:dotnet-sdk.uninstallAll",
+		"onCommand:dotnet-sdk.showAcquisitionLog",
+		"onCommand:dotnet-sdk.reportIssue"
+	],
+	"main": "./dist/extension.js",
+	"contributes": {
+		"commands": [
+			{
+				"command": "dotnet-sdk.reportIssue",
+				"title": "Report an issue with the .NET SDK Install Tool.",
+				"category": ".NET SDK Install Tool"
+			},
+			{
+				"command": "dotnet-sdk.acquire",
+				"title": "Install the .NET SDK.",
+				"category": ".NET SDK Install Tool"
+			},
+			{
+				"command": "dotnet-sdk.uninstallAll",
+				"title": "Uninstall all VS Code copies of .NET SDK.",
+				"category": ".NET SDK Install Tool"
+			}
+		],
+		"configuration": {
+			"title": ".NET SDK Install Tool",
+			"properties": {
+				"dotnetSDKAcquisitionExtension.enableTelemetry": {
+					"type": "boolean",
+					"default": true,
+					"description": "Enable Telemetry for the .NET SDK install tool."
+				},
+				"dotnetSDKAcquisitionExtension.installTimeoutValue": {
+					"type": "number",
+					"default": 240,
+					"description": "Timeout for installing .NET SDK in seconds."
+				}
+			}
+		}
+	},
+	"scripts": {
+		"vscode:prepublish": "npm run compile-all && npm install && cd ../vscode-dotnet-uninstall-library && webpack --mode production && cd ../vscode-dotnet-sdk-extension && webpack --mode production",
+		"vscode:uninstall": "node ./dist/ExtensionUninstall",
+		"compile": "npm run clean && tsc -p ./",
+		"watch": "npm run compile && tsc -watch -p ./",
+		"test": "npm run compile --silent && node ./dist/test/functional/runTest.js",
+		"clean": "rimraf dist",
+		"compile-all": "cd ../vscode-dotnet-runtime-library && npm install && npm run compile && cd ../vscode-dotnet-sdk-extension && npm install && npm run compile",
+		"lint": "tslint -c ../tslint.json '../vscode-dotnet-runtime-library/src/**/*.ts' '../vscode-dotnet-sdk-extension/src/**/*.ts'",
+		"webpack": "webpack --mode development"
+	},
+	"dependencies": {
+		"@types/chai": "^4.2.7",
+		"@types/mocha": "^8.0.3",
+		"@types/node": "12.0.0",
+		"@types/rimraf": "2.0.2",
+		"@types/vscode": "^1.41.0",
+		"chai": "^4.2.0",
+		"child_process": "^1.0.2",
+		"diff": ">=3.5.0",
+		"glob": "^7.1.5",
+		"hmac-drbg": "^1.0.1",
+		"is-online": "^8.2.1",
+		"mocha": "^8.1.3",
+		"open": "^7.0.2",
+		"rimraf": "2.6.3",
+		"shelljs": "^0.8.3",
+		"ts-loader": "^6.2.1",
+		"tslint": "^5.20.1",
+		"typescript": "3.4.5",
+		"vscode-dotnet-runtime-library": "file:../vscode-dotnet-runtime-library",
+		"vscode-dotnet-uninstall-library": "file:../vscode-dotnet-uninstall-library",
```

Comment:
Why have a separate uninstall library? Looking through the code is that this looks like it's only used at build time right? If that's the case why not just have a separate node.js script that you call `node` into?

Reply:
This library produces the node script that's called here: https://github.com/dotnet/vscode-dotnet-runtime/pull/163/files#diff-eec7f1511c1726a1883bdf93af89f838b72fd3fbf430edca98268b63123a6601R73, which hooks into VSCode's extension uninstall. It's kind of a weird hook in that it requires us to provide a script that's packed with the extension but with a separate entry point, so it's used at more than just build time.

I separated it out into a different library basically because of webpack- this way the extension and the uninstall scripts can be packed separately then inserted into the vsix together (logic for that [here](https://github.com/dotnet/vscode-dotnet-runtime/pull/163/files#diff-eec7f1511c1726a1883bdf93af89f838b72fd3fbf430edca98268b63123a6601R72) and [here](https://github.com/dotnet/vscode-dotnet-runtime/pull/163/files#diff-e29c8cf29668bc44ae83e958c39957ce6f85dcbf66cb35b56242ecbc62a7fc59R49)). I wasn't able to find a way to package them separately when they were sharing a package.json and webpack config, but I can definitely look more into this.

Reply:
Holllly hell, so VSCode's way to "hook" uninstall is to have a separate VSCode extension? That's nuts!

Having a separate extension here to enable webpack definitely feels odd. It'd be worth digging more into; I feel like this will be a huge point of confusion for repo onlookers

Reply:
Not really another extension, just another node script: https://code.visualstudio.com/updates/v1_21#_extension-uninstall-hook. I'll try to look into ways to consolidate these webpacks

Reply:
Yeah, I can't find a simple way to combine the webpack configs so I think we'll have to leave it for now. If I can figure out how  later when I have more time to look at it I'll definitely follow up with a PR to consolidate.

Reply:
Made an issue for follow up work, as I imagine there will be a significant amount: https://github.com/dotnet/vscode-dotnet-runtime/issues/165